File: gbm/javascript/gbm.js
const xgboost = require('xgboost');

// Load the training and testing data
const trainData = xgboost.DMatrix({ data: 'train.libsvm' });
const testData = xgboost.DMatrix({ data: 'test.libsvm' });

// Set the parameters for the XGBoost model
const params = {
  objective: 'reg:squarederror',
  max_depth: 3,
  eta: 0.1,
  subsample: 0.8,
  colsample_bytree: 0.8,
};

// Train the model
const numRounds = 100;
const booster = xgboost.train(params, trainData, numRounds);

// Make predictions on the test data
const predictions = booster.predict(testData);

// Evaluate the model
const labels = testData.getLabel();
let sumSquaredError = 0;
for (let i = 0; i < labels.length; i++) {
  const error = labels[i] - predictions[i];
  sumSquaredError += error * error;
}
const rmse = Math.sqrt(sumSquaredError / labels.length);
console.log(`RMSE: ${rmse}`);

File: gbm/rust/gbm.rs
use xgboost::{DMatrix, Booster};

fn main() {
    // Load the training and testing data
    let train_data = DMatrix::from_file("train.libsvm").unwrap();
    let test_data = DMatrix::from_file("test.libsvm").unwrap();

    // Set the parameters for the XGBoost model
    let params = [
        ("objective", "reg:squarederror"),
        ("max_depth", "3"),
        ("eta", "0.1"),
        ("subsample", "0.8"),
        ("colsample_bytree", "0.8"),
    ];

    // Train the model
    let num_rounds = 100;
    let model = Booster::train(&train_data, &params, num_rounds).unwrap();

    // Make predictions on the test data
    let predictions = model.predict(&test_data).unwrap();

    // Evaluate the model
    let labels = test_data.get_labels().unwrap();
    let sum_squared_error: f32 = labels
        .iter()
        .zip(predictions.iter())
        .map(|(label, prediction)| (label - prediction).powi(2))
        .sum();
    let rmse = (sum_squared_error / labels.len() as f32).sqrt();
    println!("RMSE: {}", rmse);
}
